monai/utils/profiling.py

```diff
@@ -219,15 +218,14 @@ def _is_thread_active(self):
     def _read_thread_func(self):
         """Read results from the queue and add to self.results in a thread stared by `__enter__`."""
         while self._is_parent() and self._is_thread_active():
-            try:
+            while not self.queue.empty():
                 result = self.queue.get()
```

Comment:
I'd suggest instead modifying the `get` method only and not have a loop-in-loop, so keep the `try/except` and add a `timeout` argument:

```suggestion
                result = self.queue.get(timeout=self.queue_timeout)
```

Reply:
Also a sleep value of 0.01 would be sufficient as well, it's still a long time as far as the CPU is concerned.

Reply:
@ericspod At least on my machine this does not work, raises an error. According to the Python docs your solution should work, however in my Python 3.9.16 implementation get() has no timeout:

```python
class SimpleQueue(object):
[...]
    def get(self):
        with self._rlock:
            res = self._reader.recv_bytes()
        # unserialize the data after having released the lock
        return _ForkingPickler.loads(res)
```
Switching to Queue instead of SimpleQueue would fix this, or we use the double loop as I proposed above.

Reply:
Let's use `Queue` then and take advantage of its own timeout mechanism.